jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/CheckpointManagerSuite.scala

```diff
@@ -17,12 +17,11 @@
 package ml.dmlc.xgboost4j.scala.spark
 
 import java.io.File
-import java.nio.file.Files
 
-import org.scalatest.{BeforeAndAfterAll, FunSuite}
+import org.scalatest.FunSuite
 import org.apache.hadoop.fs.{FileSystem, Path}
 
-class CheckpointManagerSuite extends FunSuite with PerTest with BeforeAndAfterAll {
+class CheckpointManagerSuite extends FunSuite with TmpFolderPerSuite with PerTest {
```

Comment:
I may miss something, why this change can address the issue?

is it just moving the beforeAll&afterAll to somewhere else?

Reply:
Before this change in CheckpointManagerSuite each test created the temp directory in /tmp (on linux) and never removed it.

(Note that PerTest neither did not nor does not now create any temp directories. It just create a SparkSession to use)

After this change CheckpointManagerSuite will include TmpFolderPerSuite .
TmpFolderPerSuite  will create a temp directory and then it will be removed.
Also after this change all temp folders are created as children of this root temp folder (see line 35 in this file)

Reply:
oh. yes, we manually created "test" dir there....